### PR TITLE
Displaying axis labels other than x, y & z in the hovering popup using hovertitle attribute

### LINKED ISSUE #2618: Axis title could also change the labels in the hovering popup
This is an enhancement suggestion.

When setting the title of an axis, such as:

```
        xaxis: {
            type: 'linear',
            zeroline: false,
            title: "hellox"  // <----
        },
```

it would make sense to have the same title as default value in the popup being shown on mouse hovering.  Currently, the default value stays "x", where it could have been "hellox" instead.

See an example here: https://codepen.io/anon/pen/qoGQvx

See also the riginal question asked on [Stack Overflow](https://stackoverflow.com/questions/49820560/how-to-change-only-the-title-of-hovering-popup-in-plotly-js?noredirect=1#comment87407776_49820560)

Please, let me know if that makes sense.

### PR DESCRIPTION
**Pull request by archmoj**
Supersedes #3317.
Fix #2618 and #265.
A new attribute `hovertitle` could be applied with each axis to enable this feature.
For example in 3D scenes if the x, y, z dimensions are presenting latitude, longitude and elevation (or other coordinates e.g. polar) this info could be displayed in the hovering popup instead of x, y and z.
@plotly/plotly_js

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
I wonder if we shouldn't just prioritize `hovertemplate` instead and then they can do this directly?

**Comment by antoinerg** [comment]
@nicolaskruchten I wonder the same thing since it should be possible to do anything with `hovertemplate` already :thinking:

File: src/components/fx/hover.js [hunk 1]
```diff
@@ -931,10 +931,24 @@ function createHoverText(hoverData, opts, gd) {
...
+            if(gd._fullLayout.yaxis) yLetter = gd._fullLayout.yaxis.hovertitle || yLetter;
+        }
+
         if(d.zLabel !== undefined) {
-            if(d.xLabel !== undefined) text += 'x: ' + d.xLabel + '<br>';
-            if(d.yLabel !== undefined) text += 'y: ' + d.yLabel + '<br>';
-            text += (text ? 'z: ' : '') + d.zLabel;
+            if(d.xLabel !== undefined) text += xLetter + ': ' + d.xLabel + '<br>';
+            if(d.yLabel !== undefined) text += yLetter + ': ' + d.yLabel + '<br>';
+            text += (text ? zLetter + ': ' : '') + d.zLabel;
```
Comment (etpinard) [inline src/components/fx/hover.js]: Hmm. So `hovertitle` for cartesian axes only has an effect for trace with a "z" field. That's ok, but should make that clear in the attribute description.

File: src/plots/polar/layout_attributes.js [hunk 2]
```diff
@@ -118,6 +118,7 @@ var radialAxisAttrs = {
     // might need a 'titleside' and even 'titledirection' down the road
 
     hoverformat: axesAttrs.hoverformat,
+    hovertitle: axesAttrs.hovertitle,
```
Comment (etpinard) [inline src/plots/polar/layout_attributes.js]: We could set a "hard" `dflt value: `'r'` for `radialaxis.hovertitle` and `'θ'` for `angularaxis.hovertitle`

File: test/image/mocks/gl3d_ibm-plot.json [hunk 3]
```diff
@@ -304,6 +304,7 @@
         "scene": {
             "xaxis": {
                 "title": "Term",
+                "hovertitle": "Term",
```
Comment (etpinard) [inline test/image/mocks/gl3d_ibm-plot.json]: Adding `hovertitle` to a few mock is ok, but we should really test this issue jasmine tests.

- https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/hover_label_test.js for cartesian 
- https://github.com/plotly/plotly.js/blob/884d2fa4cf76529bd6cb819fa2d4fec1f43fadfa/test/jasmine/tests/ternary_test.js#L120-L174 for ternary 
- https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/gl3d_plot_interact_test.js#L107 for gl3d
- https://github.com/plotly/plotly.js/blob/884d2fa4cf76529bd6cb819fa2d4fec1f43fadfa/test/jasmine/tests/scatterpolar_test.js#L83-L200 for polar
- https://github.com/plotly/plotly.js/blob/884d2fa4cf76529bd6cb819fa2d4fec1f43fadfa/test/jasmine/tests/carpet_test.js#L633-L681 for carpet

----

Oh and we should also `hovertitle` to `geo.lonaxis` and `geo.lataxis` as displayed on https://codepen.io/etpinard/pen/gqKjeB

**Comment by archmoj** [comment]
Closing now that `hovertemplate` is supported for 3D traces as well.

**Pull request closed without merging**